src/com/owncloud/android/operations/CreateShareViaLinkOperation.java

```diff
@@ -121,30 +106,35 @@ public String getPath() {
         return mPath;
     }
 
-    public ShareType getShareType() {
-        return mShareType;
-    }
-
-    public String getShareWith() {
-        return mShareWith;
-    }
-
-    public boolean getPublicUpload() {
-        return mPublicUpload;
-    }
-
     public String getPassword() {
         return mPassword;
     }
 
-    public int getPermissions() {
-        return mPermissions;
+    public Intent getSendIntent() {
+        return mSendIntent;
     }
 
-    public Intent getSendIntent() {
+    public Intent getSendIntentWithSubject(Context context) {
+        if (context != null && mSendIntent != null && mSendIntent.getStringExtra(Intent.EXTRA_SUBJECT) != null) {
+            if (getClient() == null || getClient().getCredentials().getUsername() == null) {
```

Comment:
Probably it can't happen with the current code. But I also though that in the past, until the message composition made the app crash and we had to add the alternative subject without user name. Extra check will not harm.